tools/create_spec_file.py

```diff
@@ -0,0 +1,272 @@
+
+    Parameters
+    ----------
+    *args, **kwargs
+        Positional and keyword arguments to pass to `build_img()`.
+    """
+    img = build_img(*args, **kwargs)
+    try:
+        yield img
+    finally:
+        remove_img(img)
+
+
+def create_pkg_list(img: str, env: str = "base") -> str:
+    """
+    List conda packages in a Docker image.
+
+    Parameters
+    ----------
+    img : str
+        The Docker image ID or tag.
+    env : str, optional
+        The name of the conda environment. Defaults to 'base'.
+
+    Returns
+    -------
+    str
+        The conda environment specs.
+    """
+    conda_cmd = f"conda list --name={env} --explicit"
+    args = ["docker", "run", "--rm", img, "bash", "-c", conda_cmd]
+    res = subprocess.run(args, capture_output=True, check=True, text=True)
+    return res.stdout
+
+
+def sort_pkgs(specs: str) -> str:
+    """
+    Sort the contents of a conda spec file alphabetically.
+
+    Parameters
+    ----------
+    specs : str
+        A multi-line string listing the specs of a conda environment, such as produced
+        by `conda list`. Any initial lines that begin with '#' or '@' are considered
+        header lines and their order is preserved.
+
+    Returns
+    -------
+    str
+        A copy of `specs` with the contents sorted alphabetically.
+    """
+    lines = specs.splitlines()
+
+    # Get the index of the first line that doesn't begin with '#' or '@'.
+    index = 0
+    for index, line in enumerate(lines):
+        if not line.startswith(("#", "@")):
+            break
+
+    sorted_lines = lines[:index] + sorted(lines[index:])
+    return "\n".join(sorted_lines) + "\n"
+
+
+def create_spec_file(target: str) -> None:
+    """
+    Create a new conda environment spec file.
+
+    Parameters
+    ----------
+    target : {'runtime', 'dev', 'soil-moisture'}
+        'runtime':
+          Create a spec file containing runtime dependencies of ISCE3.
+
+        'dev':
+          Create a spec file containing build-time dependencies of ISCE3.
+
+        'soil-moisture':
+          Create a spec file containing dependencies of the NISAR SWNG SoilMoisture
+          software.
+    """
+    # The base Docker image to inherit from. This should match the base image used by
+    # the NISAR ADT Docker images.
+    base_repository = "cae-artifactory.jpl.nasa.gov:16003/gov/nasa/jpl/iems/sds/infrastructure/base/jplsds-oraclelinux"
+    base_tag = "8.10.250401"
+    base_digest = "sha256:e37c210f26cfe7660808e536937fcf6ac359b3ec5713211e46519d7a35973a57"
+    base_img = f"{base_repository}:{base_tag}@{base_digest}"
+
+    # The URL of the Miniforge installer script to run. This should match the Miniforge
+    # version used by the NISAR ADT Docker image.
+    miniforge_release = "24.11.3-2"
+    miniforge_url = f"https://github.com/conda-forge/miniforge/releases/download/{miniforge_release}/Miniforge3-Linux-x86_64.sh"
```

Comment:
Pointing these out to myself as I'll need to remember to update these as our base image / miniforge release is updated ;)